.github/workflows/publish.yml

```diff
@@ -9,31 +9,18 @@ jobs:
     name: Build and push to PyPI
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Prep Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: "3.11"
 
-      - name: Install build tools
-        run: |
-          python -m pip install build setuptools twine wheel --user
-        continue-on-error: false
-      - name: Build source and wheel
-        id: build
-        run: |
-          python -m build --outdir dist/
-        continue-on-error: true
-      - name: Check the output
-        run: |
-          python -m twine check --strict dist/*
-        continue-on-error: false
-      - name: Die on failure
-        if: steps.build.outcome != 'success'
-        run: exit 1
+      - name: Build
+        run: pipx run build
+
       - name: Publish to PyPI
-        uses: pypa/gh-action-pypi-publish@master
+        uses: pypa/gh-action-pypi-publish@release/v1
         with:
           user: __token__
           password: ${{ secrets.AUTH_TOKEN }}
```

Comment:
You might wanna use trusted publishing: https://docs.pypi.org/trusted-publishers/